plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -425,6 +437,59 @@ public void testRegisterTableWithInvalidMetadataFileName()
         }
     }
 
+    @Test
+    public void testRegisterHadoopTableAndRead()
+            throws IOException
+    {
+        // create a temporary table to generate data file
+        String tempTableName = "temp_table_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tempTableName + " (id INT, name VARCHAR)");
+        assertUpdate("INSERT INTO " + tempTableName + " values(1, 'INDIA')", 1);
+        String dataFilePath = (String) computeScalar("SELECT \"$path\" FROM " + tempTableName);
+
+        // create hadoop table
+        String hadoopTableName = "hadoop_table_" + randomNameSuffix();
+        String hadoopTableLocation = metastoreDir.getPath() + "/" + hadoopTableName;
+        HadoopTables hadoopTables = new HadoopTables(newEmptyConfiguration());
+        Schema schema = new Schema(ImmutableList.of(
+                Types.NestedField.optional(1, "id", Types.IntegerType.get()),
+                Types.NestedField.optional(2, "name", Types.StringType.get())));
+        Table table = hadoopTables.create(
+                schema,
+                PartitionSpec.unpartitioned(),
+                SortOrder.unsorted(),
+                ImmutableMap.of(),
+                hadoopTableLocation);
+
+        // copy and append data file to hadoop table
+        String hadoopTableDataLocation = hadoopTableLocation + "/data";
+        Files.createDirectory(Path.of(hadoopTableDataLocation));
+        Files.copy(Path.of(dataFilePath), Path.of(hadoopTableDataLocation, "file-a.orc"));
+        DataFile fileA =
+                DataFiles.builder(PartitionSpec.unpartitioned())
+                        .withFormat(FileFormat.ORC)
+                        .withInputFile(localInput(new File(dataFilePath)))
+                        .withPath(hadoopTableDataLocation + "/file-a.orc")
+                        .withRecordCount(1)
+                        .build();
```

Comment:
```suggestion
        DataFile dataFile =
                DataFiles.builder(PartitionSpec.unpartitioned())
                        .withFormat(FileFormat.ORC)
                        .withInputFile(localInput(new File(dataFilePath)))
                        .withPath(dataFilePath)
                        .withRecordCount(1)
                        .build();
```

If you take this suggestion, remove `throws IOException` from the method definition.

Reply:
So you are referring here that we could use data files from a different location than Hadoop table location?

Reply:
Yes, Iceberg allows this.